examples/rpc.py

```diff
@@ -0,0 +1,227 @@
+        connect_participant("greeter", room_name),
+        connect_participant("math-genius", room_name),
+    )
+
+    # Register all methods for the receiving participant
+    await register_receiver_methods(greeters_room, math_genius_room)
+
+    try:
+        print("\n\nRunning greeting example...")
+        await asyncio.gather(perform_greeting(callers_room))
+    except Exception as error:
+        print("Error:", error)
+
+    try:
+        print("\n\nRunning error handling example...")
+        await perform_divide(callers_room)
+    except Exception as error:
+        print("Error:", error)
+
+    try:
+        print("\n\nRunning math example...")
+        await perform_square_root(callers_room)
+        await asyncio.sleep(2)
+        await perform_quantum_hypergeometric_series(callers_room)
+    except Exception as error:
+        print("Error:", error)
+
+    print("\n\nParticipants done, disconnecting...")
+    await callers_room.disconnect()
+    await greeters_room.disconnect()
+    await math_genius_room.disconnect()
+
+    print("Participants disconnected. Example completed.")
+
+
+async def register_receiver_methods(
+    greeters_room: rtc.Room, math_genius_room: rtc.Room
+):
+    async def arrival_method(
+        request_id: str,
+        caller_identity: str,
+        payload: str,
+        response_timeout_ms: int,
+    ):
+        print(f'[Greeter] Oh {caller_identity} arrived and said "{payload}"')
+        await asyncio.sleep(2)
+        return "Welcome and have a wonderful day!"
+
+    async def square_root_method(
+        request_id: str,
+        caller_identity: str,
+        payload: str,
+        response_timeout_ms: int,
+    ):
+        json_data = json.loads(payload)
+        number = json_data["number"]
+        print(
+            f"[Math Genius] I guess {caller_identity} wants the square root of {number}. I've only got {response_timeout_ms / 1000} seconds to respond but I think I can pull it off."
+        )
+
+        print("[Math Genius] *doing math*…")
+        await asyncio.sleep(2)
+
+        result = number**0.5
+        print(f"[Math Genius] Aha! It's {result}")
+        return json.dumps({"result": result})
+
+    async def divide_method(
+        request_id: str,
+        caller_identity: str,
+        payload: str,
+        response_timeout_ms: int,
+    ):
+        json_data = json.loads(payload)
+        dividend = json_data["dividend"]
+        divisor = json_data["divisor"]
+        print(
+            f"[Math Genius] {caller_identity} wants to divide {dividend} by {divisor}."
+        )
+
+        result = dividend / divisor
+        return json.dumps({"result": result})
+
+    await greeters_room.local_participant.register_rpc_method("arrival", arrival_method)
+    await math_genius_room.local_participant.register_rpc_method(
+        "square-root", square_root_method
+    )
+    await math_genius_room.local_participant.register_rpc_method(
+        "divide", divide_method
+    )
```

Comment:
does registering needs to be async?

Reply:
yeah this PR needs some small updates on latest FFI once the protobuf issues are resolved so i can test it. i'll clean this up when i do

Reply:
done